draft-ietf-tls-esni.md

```diff
@@ -429,6 +429,18 @@ The "ech_outer_extensions" extension is only used for compressing the
 ClientHelloInner. It MUST NOT be sent in either ClientHelloOuter or
 ClientHelloInner.
 
+Note that it is possible to implement decoding of the EncodedClientHelloInner in
+a way that creates a denial-of-service vulnerability. Specifically, the server
+needs to check that each extension in the OuterExtensions list appears in the
+ClientHelloOuter. The naive strategy would require N\*M comparisons in the worst
```

Comment:
Nit: O(N\*M) and O(N\*log(M)), since it's not exact? (And since O captures "in the worst case", you could probably even omit it then.)

Reply:
Done.